UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Services/FileShareService.cs

```diff
@@ -49,6 +49,12 @@ public byte[] GetFileData(string homeDirectoryPath, string batchId, string files
             {
                 filePaths = Directory.GetFiles(setZipPath, filesName);
             }
+            else if (!string.IsNullOrEmpty(setZipPath) && FileHelper.ValidateFilePath(setZipPath) && Directory.Exists(setZipPath)
```

Comment:
I think this if..else if..else if..else is difficult to read, especially with the similarities on the first two checks. Migth be worth converting to a switch statement as follows:

            switch (string.IsNullOrEmpty(setZipPath))
            {
                case false when FileHelper.ValidateFilePath(setZipPath) && Directory.Exists(setZipPath) && FileHelper.ValidateFilePath(Directory.GetFiles(setZipPath, filesName).FirstOrDefault()) && string.Equals("V01X01.zip", filesName, StringComparison.OrdinalIgnoreCase):
                case false when FileHelper.ValidateFilePath(setZipPath) && Directory.Exists(setZipPath) && FileHelper.ValidateFilePath(Directory.GetFiles(setZipPath, filesName).FirstOrDefault()) && (string.Equals("M01X02.zip", filesName, StringComparison.OrdinalIgnoreCase) || string.Equals("M02X02.zip", filesName, StringComparison.OrdinalIgnoreCase)):
                    filePaths = Directory.GetFiles(setZipPath, filesName);
                    break;
                default:
                {
                    if (FileHelper.ValidateFilePath(fileShareServiceConfiguration.Value.FileDirectoryPathForENC) && Directory.Exists(fileShareServiceConfiguration.Value.FileDirectoryPathForENC) && !string.Equals("README.TXT", filesName, StringComparison.OrdinalIgnoreCase))
                    {
                        filePaths = Directory.GetFiles(fileShareServiceConfiguration.Value.FileDirectoryPathForENC, string.Equals(fileType, ".TXT", StringComparison.OrdinalIgnoreCase) ? "*.TXT" : "*.000");
                    }
                    else
                    {
                        filePaths = Directory.GetFiles(fileShareServiceConfiguration.Value.FileDirectoryPathForReadme, filesName);
                    }

                    break;
                }
            }

Reply:
Updated. Thanks